### ISSUE TITLE
Future suggestion: Gnome shell extension

### ISSUE THREAD

**Issue by coxackie** [SamSchott/maestral-dropbox#68]
Running `maestral-dropbox`, using Gnome 3.34 (Arch Linux). The back-end takes up around ~55MB memory; if I also run the front-end (gui) there is an extra ~70MB of memory. Combined (~125MB) is still a bit better than running `dropbox` (>130MB), but it is clear that most memory is spent by `Qt`. For Gnome DE, there is the possibility to create extensions in JS and connect to the back-end directly; see, for example,
https://github.com/jaystrictor/gnome-shell-extension-syncthing
for the same idea applied to `syncthing`. Such extensions add almost zero memory footprint to the shell. If you would like to go the `Gtk` way at some point, maybe this would be the leanest way; and `Qt` would not be needed at all for having an indicator.

I understand that this may not be a priority right now, but I thought one should keep this in mind as a possibility.

**System (please complete the following information):**
 - Maestral version: 0.4.3
 - OS: Arch
 - Desktop environment: Gnome 3.34

**Comment by SamSchott** [comment]
Yes, I am aware that Qt accounts for more than half of the current memory usage. Also, it is by far the largest dependency to install. Unfortunately, I don't have the time at the moment to look into alternatives. But ideally I would aim for native Cocoa and Gtk 3 frontends with Qt only as a fallback.

I didn't know about the JS gnome-shell-extensions (I mostly use macOS) and will keep this in mind!
